### DISCUSSION TITLE
Error when installing contextualspellcheck

### DISCUSSION THREAD

**Discussion by galbafoz-d01** [explosion/spaCy#10761] [Help: Installation] [answered]
Hi, everyone

I cannot install contextualspellcheck due to the following errors:

`  Building wheel for editdistance (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\__init__.py -> build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\_editdistance.h -> build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\def.h -> build\lib.win-amd64-cpython-39\editdistance
      running build_ext
      building 'editdistance.bycython' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for editdistance
  Running setup.py clean for editdistance
Failed to build editdistance
Installing collected packages: editdistance, contextualspellcheck
  Attempting uninstall: editdistance
    Found existing installation: editdistance 0.6.0
    Uninstalling editdistance-0.6.0:
      Successfully uninstalled editdistance-0.6.0
  Running setup.py install for editdistance ... error
  error: subprocess-exited-with-error

  × Running setup.py install for editdistance did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      running install
      C:\Users\galba\anaconda3\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-39
      creating build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\__init__.py -> build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\_editdistance.h -> build\lib.win-amd64-cpython-39\editdistance
      copying editdistance\def.h -> build\lib.win-amd64-cpython-39\editdistance
      running build_ext
      building 'editdistance.bycython' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  WARNING: No metadata found in c:\users\galba\anaconda3\lib\site-packages
  Rolling back uninstall of editdistance
  Moving to c:\users\galba\anaconda3\lib\site-packages\editdistance-0.6.0.dist-info\
   from C:\Users\galba\anaconda3\Lib\site-packages\~ditdistance-0.6.0.dist-info
  Moving to c:\users\galba\anaconda3\lib\site-packages\editdistance\
   from C:\Users\galba\anaconda3\Lib\site-packages\~ditdistance
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> editdistance`

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @galbafoz-d01 , can you check if the solution [here](https://github.com/explosion/spaCy/discussions/10610) will work for you? As this is a third-party package, I also encourage you to check directly with [the developers](https://github.com/R1j1t/contextualSpellCheck). Also, next time please format the traceback properly in markdown (you can use triple backticks) so that it's easier to read.
